src/Sylius/Bundle/CoreBundle/Form/Type/Checkout/PaymentStepType.php

```diff
@@ -29,40 +29,29 @@ class PaymentStepType extends AbstractResourceType
     /**
      * @param string $dataClass
      * @param array $validationGroups
-     * @param ChannelContextInterface $channelContext
      */
-    public function __construct($dataClass, array $validationGroups = [], ChannelContextInterface $channelContext)
+    public function __construct($dataClass, array $validationGroups = [])
     {
         parent::__construct($dataClass, $validationGroups);
-        $this->channelContext = $channelContext;
     }
 
     /**
      * {@inheritdoc}
      */
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
-        $notBlank = new NotBlank();
-        $notBlank->message = 'sylius.checkout.payment_method.not_blank';
-
         $builder
-            ->add('paymentMethod', 'sylius_payment_method_choice', [
-                'label' => 'sylius.form.checkout.payment_method',
-                'expanded' => true,
-                'property_path' => 'lastPayment.method',
-                'channel' => $this->channelContext->getChannel(),
-                'constraints' => [
-                    $notBlank,
-                ],
-            ])
-        ;
+            ->add('payments', 'collection',[
+                'type' => 'sylius_checkout_payment', 
+                'label' => false
+            ]);
     }
 
     /**
      * {@inheritdoc}
      */
     public function getName()
     {
-        return 'sylius_checkout_payment';
+        return 'sylius_checkout_paying';
```

Comment:
I suppose it has been renamed to avoid name conflict with [this new form type](https://github.com/Sylius/Sylius/pull/5240/files#diff-4c0793b01f1de696e2fc01d51e71e544R70)? How about leaving this as it was (``paying`` seems quite weird for me) and naming this new type ``sylius_shop_checkout_payment``?